packages/wallet-sdk/src/core/type/ProviderInterface.ts

```diff
@@ -20,10 +20,10 @@ interface ProviderConnectInfo {
   readonly chainId: string;
 }
 
-// properties explicitly required by EIP-1193 spec
-// + extends EventEmitter per spec recomendation
 export interface ProviderInterface extends EventEmitter {
   request<T>(args: RequestArguments): Promise<T>;
+  disconnect(): Promise<void>;
+  setAppInfo?(appName: string, appLogoUrl: string | null): void; // legacy extension only
```

Comment:
let's remove this
```suggestion
```

Reply:
Okay but then ts is going to complain about [this](https://github.com/coinbase/coinbase-wallet-sdk/pull/1166/files/f318595ddbc737e98b84cc98a8109eb3dd626045#diff-39f5c3cd1d01da4693d4d2bffc017f605a8df261c485fb878c5ae723b0779a5aL53)

So should I just create a `LegacyProviderInterface` again that extends `ProviderInterface` and adds `setAppInfo`?

Reply:
nah. just check whether `setAppInfo` function is available within the function. we don't need to expose it in the interface. we just use it internally on one location